### DISCUSSION TITLE
How to reassemble sharded files of gguf into Jan ? (Sharded GGUF support?)

### DISCUSSION THREAD

**Discussion by chris282** [janhq/jan#2925] [Get Help] [answered]
How to reassemble sharded files of gguf into Jan ?

For example, I want to execute Mixtral-8x22B-Instruct-v0.1-GGUF from Huggingface available here : 
https://huggingface.co/MaziyarPanahi/Mixtral-8x22B-Instruct-v0.1-GGUF

The file parts are sharded like this : 
- Mixtral-8x22B-Instruct-v0.1.Q3_K_L-00001-of-00002.gguf
- Mixtral-8x22B-Instruct-v0.1.Q3_K_L-00002-of-00002.gguf

How can we execute this on Jan, is it supported or planned to be supported ?

**Comment by chris282** [comment]
Today I discovered this documentation page, but I doesn't tell how to assemble sharded parts : https://jan.ai/docs/models/manage-models

**Comment by chris282** [comment]
I have found on Jan Hub a Mistral 8X7B Instruct Q4 and it is fine, my problem is resolved, but I think that this would be a great feature into Jan to import sharded GGUF files parts.

**Comment by ux-han** [comment, marked as answer]
You can import Hugging Face model directly. Guides here: https://jan.ai/changelog/2024-04-25-llama3-command-r-hugginface#import-huggingface-models-directly
